src/test/java/org/example/jvspringbootfirstbook/controller/ShoppingCartControllerTest.java

```diff
@@ -0,0 +1,182 @@
+    private static final BigDecimal PRICE = BigDecimal.valueOf(12.12);
+    private static final String DESCRIPTION = "image";
+    private static final String COVER_IMAGE = "description";
+    private static MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void beforeAll(
+            @Autowired WebApplicationContext applicationContext
+    ) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @AfterAll
+    static void afterAll(
+            @Autowired DataSource dataSource
+    ) {
+        teardown(dataSource);
+    }
+
+    @SneakyThrows
+    private static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("db/cart/controller/delete-carts.sql")
+            );
+        }
+    }
+
+    @Test
+    @DisplayName("""
+            Update item in cart with valid input returns CartItemDto
+            """)
+    @WithMockUser(username = "user", roles = "USER")
+    @SneakyThrows
+    @Sql("classpath:db/cart/controller/add-data-for-cart-test.sql")
+    void updateItemCart_Success() {
+        //Given
+        long quantity = QUANTITY + 1;
+        CartItemUpdatedDto cartItemUpdateDto = getCartItemUpdateDto((int) quantity);
+        CartItem cartItem = getCartItem(
+                getShoppingCart(getUser()), getBook());
+        Long cartItemId = 2L;
+        cartItem.setId(cartItemId);
+
+        String jsonRequest = objectMapper.writeValueAsString(cartItemUpdateDto);
+        //When
+        MvcResult result = mockMvc.perform(
+                        put("/cart/cart-items/" + cartItemId)
+                                .content(jsonRequest)
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isOk())
+                .andReturn();
+        //Then
+        CartItemDto actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(),
+                CartItemDto.class
+        );
+        Assertions.assertEquals(quantity, actual.quantity());
+    }
+
+    @Test
+    @DisplayName("""
+            Delete item from cart with valid input
+            """)
+    @WithMockUser(username = "user", roles = "USER")
+    @SneakyThrows
+    void deleteItemFromCart_Success() {
+        CartItem cartItem = getCartItem(getShoppingCart(getUser()), getBook());
+        cartItem.setId(2L);
+        Long cartItemId = cartItem.getId();
+        mockMvc.perform(delete("/cart/cart-items/" + cartItemId)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isNoContent());
+    }
```

Comment:
After making modifications in DB you should assure that DB looks the same as before this method was called. For example: 

```suggestion
    @Sql(scripts = {"classpath:database/shoppingcart/delete_shopping_cart_with_necessities.sql",
            "classpath:database/shoppingcart/add_shopping_cart_with_necessities.sql"},
            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
    void deleteItemFromCart_ValidCartItemId_ReturnsNoContentStatus() throws Exception {
        long validCartItemId = 2L;
        mockMvc.perform(delete(BASE_ENDPOINT + "/cart-items/" + validCartItemId)
                        .contentType(MediaType.APPLICATION_JSON))
                .andExpect(status().isNoContent())
                .andReturn();
    }
```

Reply:
Same applies to other methods.